tasks/revision-dir.yml

```diff
@@ -0,0 +1,16 @@
+---
+- name: "Get latest and current revision"
+  git:
+    repo: "{{ atom_repository_url }}"
+    version: "{{ atom_repository_version }}"
+    dest: "{{ atom_path }}/{{ atom_revision_directory_latest_symlink_dir }}" 
+    clone: no
+    update: no
+  register: "atom_revision_git_output"
+
+- name: "Print revisions"
+  debug: "msg='current={{ atom_revision_git_output.before }} latest={{ atom_revision_git_output.after }}'"
```

Comment:
In my tests, even with a previous revision these two results would be the same - but I think that might be because of the other issues we encountered. Hence my wondering @mamedin if you have gotten this output with different results yet. That is, _atom_revision_git_output.before_ value via akin to what we used to keep in the **latest** file before updating, and _atom_revision_git_output.after_ being the true current (which I assume the above first task does more efficiently than shell command "git ls-remote {{ atom_repository_url }} {{ atom_repository_version }} | awk '{print $1}'"  did).

That said, I think as we discussed as long as **.after** is correct (so won't try to overwrite a previous or fallback revision directory) all is well...

Reply:
@hakamine - did you have a similar concern, I guess this relates to your refactor suggestion for _revision-dir.yml_ on August 10 above?

Reply:
@DavidHume-ArtefactualSystems when I first saw the PR code I was concerned that maybe using git with "clone: no" and "update:no" would fail if there was no local repo. But re-checking the ansible docs, they provide a usage example of git with "clone: no" and "update:no", to "just get information about the repository whether or not it has already been cloned locally."  therefore I think it should be good regarding my original concern. However, the problem you encountered seems to be a different issue.

Reply:
Probably related to:

https://github.com/artefactual-labs/ansible-atom/blob/master/tasks/basic.yml#L3-L9

Using depth=1 is known problem.